src/components/input/bl-input.ts

```diff
@@ -196,29 +210,45 @@ export default class BlInput extends FormControlMixin(LitElement) {
   }
 
   firstUpdated() {
+    this.passwordInput = this.type === 'password';
     this.setValue(this.value);
   }
 
   render(): TemplateResult {
     const invalidMessage = !this.checkValidity()
-      ? html`<p id="errorMessage" aria-live="polite" class="invalid-text">${this.validationMessage}</p>`
+      ? html`<p id="errorMessage" aria-live="polite" class="invalid-text">
+          ${this.validationMessage}
+        </p>`
+      : ``;
+    const helpMessage = this.helpText
+      ? html`<p id="helpText" class="help-text">${this.helpText}</p>`
       : ``;
-    const helpMessage = this.helpText ? html`<p id="helpText" class="help-text">${this.helpText}</p>` : ``;
     const icon = this.icon
       ? html` <bl-icon class="custom-icon" name="${this.icon}"></bl-icon>`
       : '';
     const label = this.label ? html`<label for="input">${this.label}</label>` : '';
 
+    const passwordShowHide = this.passwordInput
+      ? html` <bl-icon
+          class="password-icon"
+          @click="${this.textVisiblityToggle}"
+          name="${this.passwordVisible ? 'eye_off' : 'eye_on'}"
+        ></bl-icon>`
```

Comment:
1. Instead of making icon component clickable, we use icon-only button component (small, tertiary here) to make it accessible and consistent.
2. Since SVG icons are loaded lazily, to prevent showing an empty icon while it's loading when user toggles the button, put two separate buttons beforehand and show/hide with css. We already had similar issue on Select component before #256